python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -298,17 +298,6 @@ def __str__(self) -> str:
             )
         return partition_def_str
 
-    def __eq__(self, other):
```

Comment:
Tests in this PR exposed a bug with this function. We recently added an `end` param to time window partitions defs, and since `end` is not included in this function, this function falsely evaluated defs with different end values as equivalent.

I don't think this function is necessary though, since this class is a named tuple so it should already have an `__eq__` function that we don't need to maintain like this one.